# Before your start:
- Read the README.md file
- Comment as much as you can and use the resources (README.md file)
- Happy learning!

In [1]:
# import numpy and pandas

import numpy as np
import pandas as pd

# Challenge 1 - Independent Sample T-tests

In this challenge, we will be using the Pokemon dataset. Before applying statistical methods to this data, let's first examine the data.

To load the data, run the code below.

In [2]:
# Run this code:

pokemon = pd.read_csv('../pokemon.csv')

Let's start off by looking at the `head` function in the cell below.

In [3]:
# Your code here:

pokemon.head()

#                   Name Type 1  Type 2  Total  HP  Attack  Defense  \
0  1              Bulbasaur  Grass  Poison    318  45      49       49   
1  2                Ivysaur  Grass  Poison    405  60      62       63   
2  3               Venusaur  Grass  Poison    525  80      82       83   
3  3  VenusaurMega Venusaur  Grass  Poison    625  80     100      123   
4  4             Charmander   Fire     NaN    309  39      52       43   

   Sp. Atk  Sp. Def  Speed  Generation  Legendary  
0       65       65     45           1      False  
1       80       80     60           1      False  
2      100      100     80           1      False  
3      122      120     80           1      False  
4       60       50     65           1      False

The first thing we would like to do is compare the legendary Pokemon to the regular Pokemon. To do this, we should examine the data further. What is the count of legendary vs. non legendary Pokemons?

In [4]:
# Your code here:
pokemon['Legendary'].value_counts()


False    735
True      65
Name: Legendary, dtype: int64

There are 65 legendary VS 735 non legendary Pokemons

Compute the mean and standard deviation of the total points for both legendary and non-legendary Pokemon.

In [8]:
# Your code here:
legendary_mean= pokemon[pokemon['Legendary']==True]['Total'].mean()
legendary_std= pokemon[pokemon['Legendary']==True]['Total'].std()
non_legendary_mean= pokemon[pokemon['Legendary']==False]['Total'].mean()
non_legendary_std= pokemon[pokemon['Legendary']==False]['Total'].std()

In [9]:
legendary_mean

637.3846153846154

In [10]:
legendary_std

60.93738905315344

In [11]:
non_legendary_mean

417.21360544217686

In [12]:
non_legendary_std

106.76041745713005

The computation of the mean might give us a clue regarding how the statistical test may turn out; However, it certainly does not prove whether there is a significant difference between the two groups.

In the cell below, use the `ttest_ind` function in `scipy.stats` to compare the the total points for legendary and non-legendary Pokemon. Since we do not have any information about the population, assume the variances are not equal.

In [19]:
# Your code here:
from scipy.stats import ttest_ind

legendary=pokemon[pokemon['Legendary']==True]['Total']
non_legendary=pokemon[pokemon['Legendary']==False]['Total']
 

test_total_points= ttest_ind(legendary,non_legendary, equal_var=False)
test_total_points

Ttest_indResult(statistic=25.8335743895517, pvalue=9.357954335957446e-47)

What do you conclude from this test? Write your conclusions below.

In [20]:
# Your conclusions here:

pvalue_total_p=9.357954335957446e-47
pvalue_total_p < 0.05

True

As the pvalue is less than 0.05, we can assume that there is significant difference between legendary and non legendary pokemons

How about we try to compare the different types of pokemon? In the cell below, list the types of Pokemon from column `Type 1` and the count of each type.

In [23]:
# Your code here:
type_1 = pokemon['Type 1'].value_counts()
type_1

Water       112
Normal       98
Grass        70
Bug          69
Psychic      57
Fire         52
Rock         44
Electric     44
Dragon       32
Ground       32
Ghost        32
Dark         31
Poison       28
Fighting     27
Steel        27
Ice          24
Fairy        17
Flying        4
Name: Type 1, dtype: int64

Since water is the largest group of Pokemon, compare the mean and standard deviation of water Pokemon to all other Pokemon.

In [24]:
# Your code here:

water_type_mean= pokemon[pokemon['Type 1']=='Water']['Total'].mean()
water_type_std= pokemon[pokemon['Type 1']=='Water']['Total'].std()
other_types_mean= pokemon[pokemon['Type 1']!='Water']['Total'].mean()
other_types_std= pokemon[pokemon['Type 1']!='Water']['Total'].std()


In [25]:
water_type_mean

430.45535714285717

In [27]:
water_type_std

113.1882660643146

In [28]:
other_types_mean

435.85901162790697

In [29]:
other_types_std

121.0916823020807

Perform a hypothesis test comparing the mean of total points for water Pokemon to all non-water Pokemon. Assume the variances are equal. 

In [31]:
# Your code here:

water_type = pokemon[pokemon['Type 1']=='Water']['Total']
non_water_type = pokemon[pokemon['Type 1']!='Water']['Total']
 

test_water_non_water= ttest_ind(water_type,non_water_type, equal_var = True)

test_water_non_water

Ttest_indResult(statistic=-0.4418547448849676, pvalue=0.6587140317488793)

Write your conclusion below.

In [33]:
# Your conclusions here:
pvalue_w=0.6587140317488793
pvalue_w < 0.05


False

As pvalue_w is greater than 0.05, we accept the H0, and we can assume that there is not significant difference between the mean of total points for water Pokemon to all non-water Pokemon

# Challenge 2 - Matched Pairs Test

In this challenge we will compare dependent samples of data describing our Pokemon. Our goal is to see whether there is a significant difference between each Pokemon's defense and attack scores. Our hypothesis is that the defense and attack scores are equal. In the cell below, import the `ttest_rel` function from `scipy.stats` and compare the two columns to see if there is a statistically significant difference between them.

In [36]:
# Your code here:

from scipy.stats import ttest_rel
test_ad= ttest_rel(pokemon['Attack'], pokemon['Defense'])
test_ad

Ttest_relResult(statistic=4.325566393330478, pvalue=1.7140303479358558e-05)

Describe the results of the test in the cell below.

In [39]:
# Your conclusions here:
pvalue_ad=1.7140303479358558e-05
pvalue_ad < 0.05

True

As pvalue_ad is less than 0.05 we can reject H0 and assume there is significant difference between attack and defense mean score

We are also curious about whether therer is a significant difference between the mean of special defense and the mean of special attack. Perform the hypothesis test in the cell below. 

In [42]:
# Your code here

test_sp = ttest_rel(pokemon['Sp. Atk'], pokemon['Sp. Def'])
test_sp

Ttest_relResult(statistic=0.853986188453353, pvalue=0.3933685997548122)

Describe the results of the test in the cell below.

In [43]:
# Your conclusions here:
pvalue_sp=0.3933685997548122
pvalue_sp < 0.05

False

We accept the H0 as pvalue is greater than 0.05. There is not a signifficant difference between the mean of special defense and the mean od special attack

As you may recall, a two sample matched pairs test can also be expressed as a one sample test of the difference between the two dependent columns.

Import the `ttest_1samp` function and perform a one sample t-test of the difference between defense and attack. Test the hypothesis that the difference between the means is zero. Confirm that the results of the test are the same.

In [45]:
# Your code here:
from scipy.stats import ttest_1samp
pokemon['Difference'] = pokemon['Attack'] - pokemon['Defense']
test_a_d_1 = ttest_1samp(pokemon['Difference'], 0) 
test_a_d_1  

Ttest_1sampResult(statistic=4.325566393330478, pvalue=1.7140303479358558e-05)

We can confirm the results are the same: statistic=4.325566393330478, pvalue=1.7140303479358558e-05

# Bonus Challenge - The Chi-Square Test

The Chi-Square test is used to determine whether there is a statistically significant difference in frequencies. In other words, we are testing whether there is a relationship between categorical variables or rather when the variables are independent. This test is an alternative to Fisher's exact test and is used in scenarios where sample sizes are larger. However, with a large enough sample size, both tests produce similar results. Read more about the Chi Squared test [here](https://en.wikipedia.org/wiki/Chi-squared_test).

In the cell below, create a contingency table using `pd.crosstab` comparing whether a Pokemon is legenadary or not and whether the Type 1 of a Pokemon is water or not.

In [17]:
# Your code here:



Perform a chi-squared test using the `chi2_contingency` function in `scipy.stats`. You can read the documentation of the function [here](https://docs.scipy.org/doc/scipy-0.15.1/reference/generated/scipy.stats.chi2_contingency.html).

In [18]:
# Your code here:



Based on a 95% confidence, should we reject the null hypothesis?

In [19]:
# Your answer here:

